transformer_engine/common/fused_attn/fused_attn_max_512.cu

```diff
@@ -0,0 +1,1544 @@
+        data_ptrs.insert(std::pair<uint64_t, void *>(Q_ID, devPtrQ));
+        data_ptrs.insert(std::pair<uint64_t, void *>(K_ID, devPtrK));
+        data_ptrs.insert(std::pair<uint64_t, void *>(V_ID, devPtrV));
+        data_ptrs.insert(std::pair<uint64_t, void *>(Q_SEQLEN_ID, devActualSeqlenQ));
+        data_ptrs.insert(std::pair<uint64_t, void *>(K_SEQLEN_ID, devActualSeqlenK));
+        data_ptrs.insert(std::pair<uint64_t, void *>(MASK_VAL_ID, &negInfinity));
+
+        if (tensorType == CUDNN_DATA_FLOAT) {
+            data_ptrs.insert(std::pair<uint64_t, void *>(S_CONST_ID, &scaling_factor));
+        } else if (tensorType == CUDNN_DATA_HALF) {
+            __half cast_scaling_factor{scaling_factor};
+            data_ptrs.insert(std::pair<uint64_t, void *>(S_CONST_ID, &cast_scaling_factor));
+        } else if (tensorType == CUDNN_DATA_BFLOAT16) {
+            __nv_bfloat16 cast_scaling_factor{scaling_factor};
+            data_ptrs.insert(std::pair<uint64_t, void *>(S_CONST_ID, &cast_scaling_factor));
+        } else {
+            std::cerr << "Not supported tensorType." << std::endl;
+        }
+
+        data_ptrs.insert(std::pair<uint64_t, void *>(O_ID, devPtrO));
+
+        if (bias_type != NVTE_Bias_Type::NVTE_NO_BIAS) {
+            data_ptrs.insert(std::pair<uint64_t, void *>(B_ID, devPtrBias));
+        }
+
+        if (devPtrS != nullptr) {
+            data_ptrs.insert(std::pair<uint64_t, void *>(S_ID, devPtrS));
+        }
+
+        if (enable_dropout) {
+            data_ptrs.insert(std::pair<uint64_t, void *>(D_CONST_ID, &scale_dropout));
+        }
+
+        auto variantPack = cudnn_frontend::VariantPackBuilder()
+                               .setWorkspacePointer(workspace)
+                               .setDataPointers(data_ptrs)
+                               .build();
+        cudnnStatus_t status =
+            cudnnBackendExecute(handle, plan.get_raw_desc(), variantPack.get_raw_desc());
+
+        cudnn_frontend::throw_if([status]() { return (status != CUDNN_STATUS_SUCCESS); },
+                                 "Plan execute error", status);
+    } catch (cudnn_frontend::cudnnException &e) {
+        struct cudaDeviceProp prop;
+        NVTE_CHECK_CUDA(cudaGetDeviceProperties(&prop, 0));
+
+        // this example is only for GA100 cards (cudnn Version >= 8700) and GH100
+        // cards (cudnn Version >= 8800)
+        if (!((prop.major == 8 && prop.minor == 0) ||
+              (prop.major == 9 && prop.minor == 0 && CUDNN_VERSION >= 8800)) &&
+            (e.getCudnnStatus() == CUDNN_STATUS_ARCH_MISMATCH ||
+             e.getCudnnStatus() == CUDNN_STATUS_NOT_SUPPORTED)) {
+            std::cout << "Example is only supported for GA100 (cuDNN >= 8700) and "
+                         "GH100 (cuDNN >= 8800) GPUs"
+                      << std::endl;
+        } else {
+            std::cout << "[ERROR] Exception " << e.what() << std::endl;
+        }
```

Comment:
Those checks are deprecated since we already have requirements of 8.9.1, so I replaced those checks with `NVTE_ERROR(e.what())` in https://github.com/NVIDIA/TransformerEngine/pull/175/commits/45b0f582aeaecdbd6ddca698eff97060549c4a4e for catching the cudnn_frontend internal errors.

Besides, I replaced the `cudnn_frontend::throw_if` with `NVTE_CHECK` in the same commit to better show the error line.